http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClientWireMockTest.java

```diff
@@ -125,6 +125,28 @@ public void defaultConnectionIdleTimeout() {
         }
     }
 
+    @Test
+    public void noTlsTimeout_shouldResolveToConnectTimeout() {
+        Duration connectTimeout = Duration.ofSeconds(1);
+        try (NettyNioAsyncHttpClient client = (NettyNioAsyncHttpClient) NettyNioAsyncHttpClient.builder()
+                                                                                               .connectionTimeout(connectTimeout)
+                                                                                               .build()) {
+            assertThat(client.configuration().tlsHandshakeTimeout()).isEqualTo(connectTimeout);
+        }
+    }
+
+    @Test
+    public void tlsTimeoutConfigured_shouldHonor() {
+        Duration connectTimeout = Duration.ofSeconds(1);
+        Duration tlsTimeout = Duration.ofSeconds(3);
+        try (NettyNioAsyncHttpClient client = (NettyNioAsyncHttpClient) NettyNioAsyncHttpClient.builder()
+                                                                                               .connectionTimeout(connectTimeout)
```

Comment:
What happens if you set `tlsNegotiationTimeout` first and `connectionTimeout` second?

Reply:
Yeah, good catch! Will fixed

Reply:
Actually applying this timeout it in the setter doesn't seem to be a right way to do it because it can be called multiple times. Let me fix it